extensions/arc/src/ui/tree/miaaTreeNode.ts

```diff
@@ -0,0 +1,28 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { ResourceType } from '../../constants';
+import { TreeNode } from './treeNode';
+import { MiaaModel } from '../../models/miaaModel';
+import { ControllerModel } from '../../models/controllerModel';
+import { MiaaDashboard } from '../dashboards/miaa/miaaDashboard';
+
+/**
+ * The TreeNode for displaying a SQL Managed Instance on Azure Arc
+ */
+export class MiaaTreeNode extends TreeNode {
+
+	constructor(private _model: MiaaModel, private _controllerModel: ControllerModel) {
+		super(_model.name, vscode.TreeItemCollapsibleState.None, ResourceType.sqlManagedInstances);
+	}
+
+	public async openDashboard(): Promise<void> {
+		const miaaDashboard = new MiaaDashboard(this._controllerModel, this._model);
+		await Promise.all([
```

Comment:
Should the controller model be refreshed here too?

Reply:
Possibly - I was thinking about it but I wasn't sure if that was necessary since in order for the node to show up it'd have to have been refreshed already.

I need to do a pass through through of correctly handling stuff being stale so this will likely be a part of that.